CommandLine/emake/EnigmaCallbacks.cpp

```diff
@@ -0,0 +1,92 @@
+#include "EnigmaCallbacks.hpp"
+
+#include <iostream>
+#include <pthread.h>
+#include <unistd.h>
+
+bool CallBack::_isOutputting = false;
+std::ifstream CallBack::_outFile;
+
+void* CallBack::OutputThread(void*)
+{
+    if (_outFile)
```

Comment:
The problem is that some of the compiler stuff is using UpperCamel rather than lowerCamel for functions. Only class names should begin with a capital letter. Capitalizing function names is a C# standard that I don't agree with.